src/test/ui/type-alias-impl-trait/issue-98608.stderr

```diff
@@ -1,7 +1,7 @@
-error[E0271]: type mismatch resolving `<fn() -> impl Sized {hi} as FnOnce<()>>::Output == Box<u8>`
-  --> $DIR/issue-98608.rs:4:39
+error[E0271]: expected `fn() -> impl Sized {hi}` to be a fn item that returns `Box<u8>`, but it returns `impl Sized`
```

Comment:
Similarly here, we could refer to the function by name

    expected `hi` to return `Box<u8>`, but it returns `impl Sized`